app/controllers/vehicles_controller.rb

```diff
@@ -0,0 +1,61 @@
+class VehiclesController < ApplicationController
+  before_action :authenticate_user, only: %i[index show new create update destroy]
+
+  def index
+    @vehicles = @current_user.vehicles.all
```

Comment:
I don't think you have to call `.all` here. `@current_user.vehicles` returns all vehicles belonging to the current user